Add Analytics

dhruhin opened the pull request:
Set up event tracking on the website using Firebase Analytics.
I use JQuery to listen to buttons being clicked, if there's a more effecient way, lmk.
Start off analytics by tracking which designs people are clicking on learn more about.
If there are other extras, we need to track, add it to the doc here with extras: https://www.notion.so/c773bbe84eb2415085370204cb297b09?v=cd90a745a0e542598518916dc34d13ec
![image](https://user-images.githubusercontent.com/4166009/79703065-2525ea80-825e-11ea-98c0-653304ee4d4c.png)

dhruhin committed:
Add Analytics w/ Sample Event

Set up event tracking on the website using Firebase Analytics.
I use JQuery to listen to buttons being clicked, if there's a more effecient way, lmk.
Start off by tracking which designs people are clicking on learn more about.
If there are other extras, we need to track, add it to the doc here with extras: https://www.notion.so/c773bbe84eb2415085370204cb297b09?v=cd90a745a0e542598518916dc34d13ec

jpbesgen reviewed (changes requested):
Thanks Dhruhin, this looks great!

dhruhin committed:
Add More Log Events

+ Use constants for analytics event names
+ Add is authenticated to every extra
+ Track card link being clicked
+ Track card attachment being clicked

dhruhin committed:
Track link names that are clicked

* Add a name to link/attachment events

amadrzyk commented on js/components/DesignCardComponent.js:
```diff
@@ -123,7 +123,7 @@ class DesignCardComponent extends Component {
         if(design.attachments != null && design.attachments.length > 0) {
             downloads = `Attachments: `
             design.attachments.forEach((attachment) => {
-                downloads += `<a href="${attachment.url}" target="_blank" download> ${attachment.name}</a>`;
```
Super minor, but do you mind adding `rel="noopener noreferrer"` to the links? Recently found out about this vulnerability 😬 

https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/

dhruhin replied:
Of course, added this change, landing it

amadrzyk reviewed (approved):
looks great to me, thanks Dhru! 💯

dhruhin committed:
Alex's Vulnerability

that he suggeseted
